# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/compiler/translator/ResourcesHLSL.cpp
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/compiler/translator/ResourcesHLSL.cpp
--- a/gfx/angle/checkout/src/compiler/translator/ResourcesHLSL.cpp	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/compiler/translator/ResourcesHLSL.cpp	Sat Sep 07 13:38:36 2019 +0000
@@ -385,8 +385,9 @@
                                               const int binding,
                                               const unsigned int registerIndex)
 {
-    out << "uniform RWByteAddressBuffer " << getAtomicCounterNameForBinding(binding)
-        << " : register(u" << registerIndex << ");\n";
+    // Atomic counter memory access is not incoherent
+    out << "uniform globallycoherent RWByteAddressBuffer "
+        << getAtomicCounterNameForBinding(binding) << " : register(u" << registerIndex << ");\n";
 }
 
 void ResourcesHLSL::uniformsHeader(TInfoSinkBase &out,
@@ -591,7 +592,8 @@
         out << "    struct ImageMetadata\n"
                "    {\n"
                "        int layer;\n"
-               "        int3 padding;\n"
+               "        uint level;\n"
+               "        int2 padding;\n"
                "    };\n";
 
         if (mReadonlyImageCount > 0)